src/pages/product-listing/ProductListing.jsx

```diff
@@ -23,37 +31,54 @@ const ProductListing = () => {
       priceRange ? Number(prod.price) <= Number(priceRange) : true
     );
 
-  const sortProds = (prods) => {
+  const ratingRangeFilter = (data) =>
+    data.filter((prod) =>
+      ratingRange ? Number(prod.rating) >= Number(ratingRange) : true
+    );
+
+  const categoryFilter = (data) =>
+    data.filter((prod) =>
+      categoryName.length === 0
+        ? true
+        : categoryName.includes(prod.categoryName)
+    );
+
+  // Sort function
+  const sortProds = (products) => {
     if (priceSort) {
       if (priceSort === 'LOW_TO_HIGH') {
-        return prods.sort((a, b) => a.price - b.price);
+        return products.sort((a, b) => a.price - b.price);
```

Comment:
I would suggest using the `spread syntax[ ...]` here since array.sort is a destructive method

Reply:
Noted. And changed it.